app/models/group_assignment.rb

```diff
@@ -10,6 +10,24 @@ class GroupAssignment < ApplicationRecord
   scope :ongoing, (-> { where('group_assignments.period_end > ?', Time.zone.today) })
   scope :no_end, (-> { where(period_end: nil) })
 
+
+  scope :start_within, ->(date_range) { start_before(date_range.last).start_after(date_range.first) }
+  scope :end_within, ->(date_range) { where(period_end: date_range) }
+  scope :end_after, ->(date) { where('period_end > ?', date) }
+  scope :end_before, ->(date) { where('period_end < ?', date) }
+
+  scope :start_after, ->(date) { where('period_start > ?', date) }
+  scope :start_before, ->(date) { where('period_start < ?', date) }
+
+  scope :not_end_before, ->(date) { where('period_end > ?', date) }
+
+  scope :active_between, lambda { |start_date, end_date|
+    no_end.start_before(end_date)
+          .or(
```

Comment:
Could it also need `no_start.end_after(start_date)`? If yes, `:no_start` scope is not defined yet.

Reply:
Same as above. No start means never active